WalletWasabi.Gui/Tabs/WalletManager/LoadWalletViewModel.cs

```diff
@@ -631,7 +631,16 @@ public async Task LoadWalletAsync()
 
 				try
 				{
-					var walletService = await Task.Run(async () => await Global.CreateWalletServiceAsync(keyManager));
+					while (!Global.InitializationCompleted)
+					{
+						if (Global.StoppingCts.IsCancellationRequested)
```

Comment:
It is enough to check the `InitializationCompleted` as it is set to true in any case. So you don't have to expose the CancellationTokenSource.

Reply:
It isn't. Look at the global's dispose.

Reply:
`WaitForInitializationCompletedAsync()` is elegant. ACK!